spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkPositionDeltaWrite.java

```diff
@@ -557,14 +626,16 @@ private abstract static class DeleteAndDataDeltaWriter extends BaseDeltaWriter {
 
     DeleteAndDataDeltaWriter(
         Table table,
+        Map<String, DeleteFileSet> dataFileToFileScopedDeletes,
         SparkFileWriterFactory writerFactory,
         OutputFileFactory dataFileFactory,
         OutputFileFactory deleteFileFactory,
         Context context) {
       this.delegate =
           new BasePositionDeltaWriter<>(
               newDataWriter(table, writerFactory, dataFileFactory, context),
-              newDeleteWriter(table, writerFactory, deleteFileFactory, context));
+              newDeleteWriter(
```

Comment:
We may want to put this behind a spark conf which defaults to loading the previous deletes and doing the merging in the write. Maybe it's better to handle such a conf when building the mapping (if the conf is false, we could return a empty map or don't perform the broadcast).

That way in case users hit some unforseen issue with merging they can disable the behavior dynamically.

Didn't want to introduce more configuration knobs initially to keep it simple but I can see an argument that it's worth it just to have a lever for unforseen issues with performing minor compactions as part of writes.

CC @aokolnychyi @rdblue @nastra

Reply:
Some of the tests in TestRewritePositionDeleteFilesProcedure are expectedly failing after this change since those tests have assertions on expected delete files after performing a set of delete operations but now since there's maintenance happening as part of the write, the number of delete files will be expectedly reduced.
If we decide to add a conf, that can be set otherwise I'll go ahead and update the expected number of delete files.

Reply:
I went ahead with adding the configuration and table property similar to delete file granularity.